VLDofusBotApp/src/main/kotlin/fr/lewon/dofus/bot/scripts/tasks/impl/harvest/HarvestResourceTask.kt

```diff
@@ -0,0 +1,75 @@
+package fr.lewon.dofus.bot.scripts.tasks.impl.harvest
+
+import fr.lewon.dofus.bot.core.logs.LogItem
+import fr.lewon.dofus.bot.scripts.harvest.JobSkillsManager
+import fr.lewon.dofus.bot.scripts.tasks.BooleanDofusBotTask
+import fr.lewon.dofus.bot.scripts.tasks.impl.fight.FightTask
+import fr.lewon.dofus.bot.sniffer.model.messages.game.context.GameEntitiesDispositionMessage
+import fr.lewon.dofus.bot.sniffer.model.messages.game.inventory.items.ObjectQuantityMessage
+import fr.lewon.dofus.bot.util.game.InteractiveUtil
+import fr.lewon.dofus.bot.util.game.RetryUtil
+import fr.lewon.dofus.bot.util.io.WaitUtil
+import fr.lewon.dofus.bot.util.network.info.GameInfo
+
+class HarvestResourceTask(
+    private val harvestJob : String,
+    private val stopIfNoResourcePresent: Boolean = false
+) : BooleanDofusBotTask() {
+
+    override fun doExecute(logItem: LogItem, gameInfo: GameInfo): Boolean {
+        gameInfo.eventStore.clear()
+        val couldHarvest = RetryUtil.tryUntilSuccess({ harvestAnyResource(logItem, gameInfo, harvestJob) }, 20, {
+            if (stopIfNoResourcePresent && gameInfo.interactiveElements.any { it.enabledSkills.isEmpty() }) {
+                error("No resource on map")
+            }
+            WaitUtil.waitUntil({ gameInfo.interactiveElements.any { it.enabledSkills.isNotEmpty() } }, 60000)
```

Comment:
Is the goal to wait for 60 seconds for a resource to respawn ? What's the usecase ?